modules/xcm-bridge-hub-router/src/lib.rs

```diff
@@ -0,0 +1,400 @@
+// message fee using fee factor.
+impl<T: Config<I>, I: 'static> ExporterFor for Pallet<T, I> {
+	fn exporter_for(
+		network: &NetworkId,
+		_remote_location: &InteriorMultiLocation,
+		message: &Xcm<()>,
+	) -> Option<(MultiLocation, Option<MultiAsset>)> {
+		// ensure that the message is sent to the expected bridged network
+		if *network != T::BridgedNetworkId::get() {
+			return None
+		}
+
+		// compute fee amount. Keep in mind that this is only the bridge fee. The fee for sending
+		// message from this chain to child/sibling bridge hub is determined by the
+		// `Config::ToBridgeHubSender`
+		let message_size = message.encoded_size();
+		let message_fee = (message_size as u128).saturating_mul(T::ByteFee::get());
+		let fee_sum = T::BaseFee::get().saturating_add(message_fee);
+		let fee_factor = Self::delivery_fee_factor();
+		let fee = fee_factor.saturating_mul_int(fee_sum);
+
+		log::info!(
+			target: LOG_TARGET,
+			"Going to send message ({} bytes) over bridge. Computed bridge fee {} using fee factor {}",
+			fee,
+			fee_factor,
+			message_size,
+		);
+
+		Some((T::SiblingBridgeHubLocation::get(), Some((T::FeeAsset::get(), fee).into())))
+	}
+}
+
+// This pallet acts as the `SendXcm` to the sibling/child bridge hub instead of regular
+// XCMP/DMP transport. This allows injecting dynamic message fees into XCM programs that
+// are going to the bridged network.
+impl<T: Config<I>, I: 'static> SendXcm for Pallet<T, I> {
+	type Ticket = (u32, <T::ToBridgeHubSender as SendXcm>::Ticket);
+
+	fn validate(
+		dest: &mut Option<MultiLocation>,
+		xcm: &mut Option<Xcm<()>>,
+	) -> SendResult<Self::Ticket> {
+		// we won't have an access to `dest` and `xcm` in the `delvier` method, so precompute
+		// everything required here
+		let message_size = xcm
+			.as_ref()
+			.map(|xcm| xcm.encoded_size() as _)
+			.ok_or(SendError::MissingArgument)?;
+
+		// bridge doesn't support oversized/overweight messages now. So it is better to drop such
+		// messages here than at the bridge hub. Let's check the message size.
+		if message_size > HARD_MESSAGE_SIZE_LIMIT {
+			return Err(SendError::ExceedsMaxMessageSize)
+		}
+
+		// just use exporter to validate destination and insert instructions to pay message fee
+		// at the sibling/child bridge hub
+		//
+		// the cost will include both cost of: (1) to-sibling bridg hub delivery (returned by
+		// the `Config::ToBridgeHubSender`) and (2) to-bridged bridge hub delivery (returned by
+		// `Self::exporter_for`)
+		ViaBridgeHubExporter::<T, I>::validate(dest, xcm)
+			.map(|(ticket, cost)| ((message_size, ticket), cost))
+	}
+
+	fn deliver(ticket: Self::Ticket) -> Result<XcmHash, SendError> {
+		// use router to enqueue message to the sibling/child bridge hub. This also should handle
+		// payment for passing through this queue.
+		let (message_size, ticket) = ticket;
+		let xcm_hash = T::ToBridgeHubSender::deliver(ticket)?;
```

Comment:
@svyatonik 
I dont know but I have a feeling that we should better use here for deliver the same thing what we used for validate,
I know that `ToBridgeHubSender` is Router of SovereignPaidRemoteExporter, but e.g. we could have a problem when somebody change/extend `SovereignPaidRemoteExporter::deliver` with some logic.
```suggestion
		let xcm_hash = ViaBridgeHubExporter::<T, I>::deliver(ticket)?;
```

suggestion is part of https://github.com/paritytech/parity-bridges-common/pull/2314